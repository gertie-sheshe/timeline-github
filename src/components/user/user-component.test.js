import React from 'react';
import { render, cleanup } from '@testing-library/react';
import UserComponent from './user-component';

afterEach(cleanup);

describe('User Component', () => {
    it('Matches Loader Snapshot', () => {
        let userRepoData = {
            dates: ['2020', '2021'], data: {
                '2020': [{ id: 1, name: 'test-repo-one', language: 'HTML', description: 'Test Repo' }],
                '2021': [{ id: 2, name: 'test-repo-two', language: 'Javascript', description: 'Test Repo' }]
            }
        }
        const { asFragment } = render(<UserComponent userRepos={userRepoData} />);
    
        expect(asFragment(<UserComponent />)).toMatchSnapshot();
    });
});
