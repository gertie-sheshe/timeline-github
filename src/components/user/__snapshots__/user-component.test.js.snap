// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User Component Matches Loader Snapshot 1`] = `
<DocumentFragment>
  <div
    class="user-component"
  >
    <div
      class="wrapper"
    >
      <div
        class="box-area"
      >
        <div
          class="custom"
        />
        <div
          class="box-date"
        >
          <span>
            <div />
          </span>
        </div>
        <div
          class="box-text"
        >
          <p
            class="repo-year"
          >
            July 2020
          </p>
          <a
            class="repo-title"
          >
            test-repo-one
          </a>
          <p
            class="repo-description"
          >
            Test Repo
          </p>
          <p
            class="repo-description"
          >
            Language: HTML
          </p>
        </div>
      </div>
      <div
        class="box-area"
      >
        <div
          class="custom"
        />
        <div
          class="box-date"
        >
          <span>
            <div />
          </span>
        </div>
        <div
          class="box-text"
        >
          <p
            class="repo-year"
          >
            July 2021
          </p>
          <a
            class="repo-title"
          >
            test-repo-two
          </a>
          <p
            class="repo-description"
          >
            Test Repo
          </p>
          <p
            class="repo-description"
          >
            Language: Javascript
          </p>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
